syntax = "proto3";

package helloworld;

service Greeter {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc SayHelloStream (stream HelloRequest) returns (stream HelloReply) {}
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}


// syntax = "proto3";
// 
// package services;
// 
// message HelloRequest {
//  string name = 1;
// }
// 
// message HelloReply {
//  string message = 1;
// }
// 
// message SumRequest {
//   string a = 1;
//   string b = 2;
// }
// 
// message SumReply {
//   string result = 1;
// }
// 
// service ServiceA {
//   rpc SayHello (HelloRequest) returns (HelloReply) {}
//    rpc GetSum (SumRequest) returns (SumReply) {}
// }
// 
// message EchoRequest {
//   string echo = 1;
// }
// 
// message EchoReply {
//   string echo = 1;
// }
// 
// service ServiceB {
//   rpc Echo (EchoRequest) returns (EchoReply) {}
// }